tests/everflow/everflow_test_utilities.py

```diff
@@ -303,10 +301,96 @@ def acl_stage(self):
         Get the ACL stage for this set of test cases.
 
         Used to parametrize test cases based on the ACL stage.
-
         """
         pass
 
+    def send_and_check_mirror_packets(self,
+                                      setup,
+                                      mirror_session,
+                                      ptfadapter,
+                                      duthost,
+                                      mirror_packet,
+                                      src_port=None,
+                                      dest_ports=None,
+                                      expect_recv=True):
+        expected_mirror_packet = self._get_expected_mirror_packet(mirror_session,
+                                                                  setup,
+                                                                  duthost,
+                                                                  mirror_packet)
+
+        if not src_port:
+            src_port = self._get_random_src_port(setup)
+
+        if not dest_ports:
+            dest_ports = [self._get_monitor_port(setup, mirror_session, duthost)]
+
+        ptfadapter.dataplane.flush()
+        testutils.send(ptfadapter, src_port, mirror_packet)
+
+        if expect_recv:
+            _, received_packet = testutils.verify_packet_any_port(
+                ptfadapter,
+                expected_mirror_packet,
+                ports=dest_ports
+            )
+
+            logging.info("received: %s", packet.Ether(received_packet).summary())
+
+            inner_packet = self._extract_mirror_payload(received_packet, len(mirror_packet))
+            logging.info("inner_packet: %s", inner_packet.summary())
+            logging.info("expected_packet: %s", mirror_packet.summary())
+            pytest_assert(Mask(inner_packet).pkt_match(mirror_packet), "Mirror payload does not match received packet")
+        else:
+            testutils.verify_no_packet_any(ptfadapter, expected_mirror_packet, dest_ports)
+
+    def _get_expected_mirror_packet(self, mirror_session, setup, duthost, mirror_packet):
+        payload = mirror_packet.copy()
+
+        # Add vendor specific padding to the packet
+        if duthost.facts["asic_type"] in ["mellanox"]:
+            payload = binascii.unhexlify("0" * 44) + str(payload)
+
+        if duthost.facts["asic_type"] in ["barefoot"]:
+            payload = binascii.unhexlify("0" * 24) + str(payload)
+
+        # If the mirroring type is egress, we expect the TTL to drop by 1
+        expected_ttl = int(mirror_session["session_ttl"]) - (self.mirror_type == "egress")
+
+        expected_packet = testutils.simple_gre_packet(
+            eth_src=setup["router_mac"],
+            ip_src=mirror_session["session_src_ip"],
+            ip_dst=mirror_session["session_dst_ip"],
+            ip_dscp=int(mirror_session["session_dscp"]),
+            ip_id=0,
```

Comment:
what we are specifying ip_id ? we don't care about it.

Reply:
Honestly I am not sure, I copied this from the ansible test case 😅. Let me take it out and see what happens.

Reply:
I have no idea why but this seems to be required. :shrug: